### PR TITLE
feat(upstream): add upstream priority field

### LINKED ISSUE #2192: Dashboard  upstream does not display the priority field, and the priority setting will be lost when click the commit button
APISIX的路由是支持通过设置priority来满足不同的转发优先级的，但是这个特性并没有在dashboard上开放设置入口，因此我们是通过admin api 来设置的。当通过admin api 设置好priority之后，在dashboard的上游查看界面中，却不显示priority字段，一旦在这个界面点击提交，priority字段设置会丢失，导致既定优先级失效，可能引发不必要的现网故障。

Admin Api 设置的upstream：
image

在dashboard 的upstream view中显示如图：
image

一旦在这个界面点击了提交，这个字段将会被删除。

这里推荐2个解决方案：
1、dashboard支持下priority设置特性；
2、dashboard暂时不能支持的功能特性，建议优先采用合并逻辑，而不是覆写，至少不要冲掉已存在的元数据（推荐）

### PR DESCRIPTION
**Pull request by TakumiInui5432**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Add upstream priority support.

**Related issues**

fix #2192 

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by zaunist** [comment]
support priority now

image

**Comment by zaunist** [comment]
Hi, community. I have add a field to `node`, and I already update backend code and E2E test case, but this change will effect frontend, so I need someone to help me to update frontend code. @juzhiyuan @guoqqqi

**Comment by juzhiyuan** [comment]
> Hi, community. I have add a field to `node`, and I already update backend code and E2E test case, but this change will effect frontend, so I need someone to help me to update frontend code. @juzhiyuan @guoqqqi

Hi @zaunist, just contacted with @guoqqqi, he will help you to do this :)

**Comment by zaunist** [comment]
For support FQDN https://github.com/apache/apisix-dashboard/issues/2080, we can't support edit `priority` on page, only support edit in editor.

image

**Comment by zaunist** [comment]
cc @nic-chen @bzp2010 @starsz

**Comment by guoqqqi** [comment]
need to fix the e2e test

**Comment by zaunist** [comment]
cc @guoqqqi

**Comment by guoqqqi** [comment]
It looks like we can now only configure the priority field via the editor, and we may need to update the node configuration of the upstream configuration form:
How about adding a type selection for the configuration upstream form node? For example, when selecting the FQDN type, only localhost and weight are displayed; otherwise, all are displayed.

Of course, we can do it in another PR if we need to. What do others think of it? 😄

File: api/test/e2enew/upstream/upstream_priority_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	})
+	ginkgo.It("delete upstream", func() {
+		base.RunTestCase(base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/upstreams/priority",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		})
+	})
```
Comment (starsz) [inline api/test/e2enew/upstream/upstream_priority_test.go:63]: Hi, Can you add test to cover the feature of priority?

Comment (zaunist) [reply api/test/e2enew/upstream/upstream_priority_test.go:63]: OK

Comment (zaunist) [reply api/test/e2enew/upstream/upstream_priority_test.go:63]: Added test case

File: api/internal/core/store/validate.go [hunk 2]
```diff
@@ -165,12 +165,6 @@ func checkUpstream(upstream *entity.UpstreamDef) error {
 		return nil
 	}
 
-	if upstream.PassHost == "node" && upstream.Nodes != nil {
-		if nodes := entity.NodesFormat(upstream.Nodes); len(nodes.([]*entity.Node)) != 1 {
-			return fmt.Errorf("only support single node for `node` mode currently")
-		}
-	}
```
Comment (starsz) [inline api/internal/core/store/validate.go:172]: @guoqqqi Need to adjust the change.

**Review by starsz — APPROVED** [review]

File: api/internal/core/store/validate.go (hunk 2 above)
Comment (nic-chen) [reply api/internal/core/store/validate.go:172]: should not remove this block of code. see: https://github.com/apache/apisix/blob/master/apisix/upstream.lua#L397-L401

Comment (zaunist) [reply api/internal/core/store/validate.go]: fixed

**Review by juzhiyuan — APPROVED** [review]

**Comment by juxianzheng** [comment]
> 看来我们现在只能通过编辑器配置priority字段了，可能需要更新上游配置表单的节点配置： 为配置上游表单节点添加一个类型选择怎么样？比如选择FQDN类型时，只显示localhost和权重；否则，全部显示。
> 
> 当然，如果需要，我们可以在另一个 PR 中进行。其他人怎么看？😄



> 查看我们现在可以通过编辑器配置优先级字段了，可能需要更新配置表单的节点配置： 为配置启动表单节点添加一个类型选择吗？比如选择FQDN类型时，只显示本地主机和权限重；否则，全部显示。
> 
> 当然，如果需要，我们可以在另一个公关中进行。其他人怎么看？😄

I think it can

**Review by nic-chen — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**